packages/shopping/public/product.js

```diff
@@ -0,0 +1,32 @@
+/*global templates, util, alert, api, DOMPurify, $*/
+'use strict';
+
+$(function() {
+  const id = util.getProductId();
+  if (id) {
+    api.getProduct(id, function(product) {
+      if (product) {
+        const productHtml = templates.details
+          .replace(/#NAME#/gi, DOMPurify.sanitize(product.name))
+          .replace(
+            /#PRICE#/gi,
+            new Intl.NumberFormat('en-US', {
+              maximumSignificantDigits: 3,
+            }).format(DOMPurify.sanitize(product.price)),
+          )
+          .replace(/#UNFORMATTED-PRICE#/g, DOMPurify.sanitize(product.price))
+          .replace(/#IMAGE#/gi, DOMPurify.sanitize(product.image))
+          .replace(/#DESCRIPTION#/gi, DOMPurify.sanitize(product.description))
```

Comment:
This is super brittle. How can we check that `DOMPurify.sanitize` was called in all places where it's required? How to ensure we don't forget to call `DOMPurify.sanitize` when adding more features in the future?

I think it would be much easier to switch to Handlebars as I mentioned earlier, rather than adding more hacks to the already hacky template rendering. Anyhow 🤷‍♂️

Reply:
Let's switch to Handlebars in https://github.com/strongloop/loopback4-example-shopping/issues/478#issuecomment-574176625.